Tests/ApolloTests/RequestChainTests.swift

```diff
@@ -963,4 +963,136 @@ class RequestChainTests: XCTestCase {
     requestChain = nil
     XCTAssertNil(weakRequestChain)
   }
+
+  // MARK: `proceedAsync` Tests
+
+  struct SimpleForwardingInterceptor_deprecated: ApolloInterceptor {
+    var id: String = UUID().uuidString
+
+    let expectation: XCTestExpectation
+
+    func interceptAsync<Operation>(
+      chain: Apollo.RequestChain,
+      request: Apollo.HTTPRequest<Operation>,
+      response: Apollo.HTTPResponse<Operation>?,
+      completion: @escaping (Result<Apollo.GraphQLResult<Operation.Data>, Error>) -> Void
+    ) {
+      expectation.fulfill()
+
+      chain.proceedAsync(request: request, response: response, completion: completion)
+    }
+  }
+
+  struct SimpleForwardingInterceptor: ApolloInterceptor {
+    var id: String = UUID().uuidString
+
+    let expectation: XCTestExpectation
+
+    func interceptAsync<Operation>(
+      chain: Apollo.RequestChain,
+      request: Apollo.HTTPRequest<Operation>,
+      response: Apollo.HTTPResponse<Operation>?,
+      completion: @escaping (Result<Apollo.GraphQLResult<Operation.Data>, Error>) -> Void
+    ) {
+      expectation.fulfill()
+
+      chain.proceedAsync(
+        request: request,
+        response: response,
+        interceptor: self,
+        completion: completion
+      )
+    }
+  }
+
+  func test__proceedAsync__givenInterceptors_usingDeprecatedFunction_shouldCallAllInterceptors() throws {
+    let expectations = [
+      expectation(description: "Interceptor 1 executed"),
+      expectation(description: "Interceptor 2 executed"),
+      expectation(description: "Interceptor 3 executed")
+    ]
+
+    let requestChain = InterceptorRequestChain(interceptors: [
+      SimpleForwardingInterceptor_deprecated(expectation: expectations[0]),
+      SimpleForwardingInterceptor_deprecated(expectation: expectations[1]),
+      SimpleForwardingInterceptor_deprecated(expectation: expectations[2])
+    ])
+
+    let request = JSONRequest(
+      operation: MockQuery<Hero>(),
+      graphQLEndpoint: TestURL.mockServer.url,
+      clientName: "test-client",
+      clientVersion: "test-client-version"
+    )
+
+    // when
+    requestChain.kickoff(request: request) { result in }
```

Comment:
We don't care about the response in these tests, there are other tests for that. These purely ensure that the interceptors are executed in the correct order.